src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,93 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Locks the current handle and reads a line of input, returning a `String` containing
+/// the input.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// [`Stdin::lock`]: struct.Stdin.html#method.lock
+/// [`Stdout::lock`]: struct.Stdout.html#method.lock
+///
+/// ### Note: Safety and Usage
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+/// 
+/// [`Stdin::read_line`]: struct.Stdin.html#method.read_line
+///
+/// # Examples
+///
+/// ```rust
+/// use std::io::inputln;
+///
+/// fn main() {
+///     let user_input = inputln("Please enter some text: ");
+///         
+///     println!("You typed: {}", user_input);
+///         
+/// }
+/// ```
+pub fn inputln(prompt: &str) -> String {
```

Comment:
The suffix "ln" in `inputln()` doesn't capture the difference to `input()`: In case of `print!` vs `println!`, the `ln` represents the newline which is printed at the end. Here the difference is that a message is printed, not a newline. Maybe `input_prompt` could fit.

Reply:
I changed the function name to ```input_prompt(&str)``` to avoid confusion.